Index: src/splashy_video.c
===================================================================
--- src/splashy_video.c.orig
+++ src/splashy_video.c
@@ -1144,6 +1144,8 @@ splashy_start_splash ()
          */
          DirectFBSetOption ("disable-module", "linux_input");
          
+	/* segfaults with 1.2.3 */
+	DirectFBSetOption ("disable-module", "x11input");
 
         /*
          * TODO doesn't solve anything: DirectFBSetOption ("dont-catch",
